presto-main/src/main/java/com/facebook/presto/type/DoubleOperators.java

```diff
@@ -238,6 +239,28 @@ public static long hashCode(@SqlType(StandardTypes.DOUBLE) double value)
     }
 
     @ScalarOperator(SATURATED_FLOOR_CAST)
+    @SqlType(StandardTypes.FLOAT)
+    public static long saturatedFloorCastToFloat(@SqlType(StandardTypes.DOUBLE) double value)
+    {
+        float result;
+        float minFloat = -1.0f * Float.MAX_VALUE;
+        if (value <= minFloat) {
+            result = minFloat;
+        }
+        else if (value >= Float.MAX_VALUE) {
+            result = Float.MAX_VALUE;
+        }
+        else {
+            result = (float) value;
```

Comment:
I can add `strictfp` modifier to this method. 

By my understanding, if the `strictfp` is set: 
- All the `float` values are being coerced to `double` for any operation on `float` and `double`
- Any `float` value can be represented precisely as `double`

So. lets follow this method line by line, considering that `strictfp` is set:

```
        float result;
        float minFloat = -1.0f * Float.MAX_VALUE;
        // minFloat is coerced to double precisely. So, this comparison is correct.
        if (value <= minFloat) {
            result = minFloat;
        }
```

```
       else if (value >= Float.MAX_VALUE) {
             // Float.MAX_VALUE is coerced to double precisely. So, this comparison is correct.
            result = Float.MAX_VALUE;
        }
```

```
        else {
            // downcasting to float. Cast can return exact value, or the closest value (either "higher" or "lower")
            result = (float) value;
           // cast result back to double (this is supposed to be precise), check if the value is returned is "higher"
            if (result > value) {
                // if it is - than decrease it to "lower"
                result = Math.nextDown(result);
            }
            checkState(result <= value);
        }
```

Is everything what i said correct? If so, i'm just going to add this modifier. Please correct me if i'm wrong in some of my assumptions.